api/src/rest/views/entity.py

```diff
@@ -21,6 +21,7 @@ class EntityCAView(generics.ListAPIView):
     queryset = models.Entity.objects.all()
     serializer_class = serializers.EntityBasicSerializer
 
+
 class EntityByCN(generics.RetrieveAPIView):
```

Comment:
Gdzie jest to używane?

Jeśli dobrze rozumiem co to ma być to od tego są filtry:

```python

class ComponentFilter(filters.FilterSet):
    """
    Component list filter for AJAX API.
    """
    class Meta:
        """
        Filter metadata.
        """
        model = Component
        fields = ['id', 'author', 'since', 'conflict', 'part', 'vendor',
                  'product', 'version', 'language']

```python
class ComponentViewSet(viewsets.ModelViewSet):
    """
    AJAX API Component ViewSet with specialized serializers per action.
    """
    queryset = Component.objects.all()
    filterset_class = ComponentFilter
    serializers = {
        ('list', 'destroy',): ComponentListSerializer,
        ('create', 'update', 'retrieve',): ComponentDetailSerializer
    }
    # TODO alias
    search_fields = ('cpe', 'description', 'author__username',)
    filter_backends = (
        filters.DjangoFilterBackend,
        rest_filters.SearchFilter,
        rest_filters.OrderingFilter,
    )
    ordering = ('-id',)
```

Mogłem coś pominąć przy kopiowaniu bo tak naprawdę przygotowaliśmy swój BaseViewSet i BaseFilter które zawierają wspólne rzeczy dla wszystkich widoków

Reply:
W jakim stopniu te ogólne widoki (jak listowanie/sortowanie/filtrowanie wszystkich entity) są potrzebne? Czy wystarcyz panel administratora django? Czy dostęp do tych endpointów/panelu ma być chroniony przez inne zbiór certyfikatów niż klienckie?

Reply:
Ze względu na różnice w działaniu RestFramework i czystego django raczej będziemy odchodzili od wykorzystywania wbudowanego panelu admina na rzecz API + ReactAdmin

To nie jest jednak coś co leży w zakresie twojego projektu.

Stąd moje pytanie. Jakie jest zastosowanie tego endpoint-a?

Reply:
Miał być użyty przy dodawaniu danych certyfikatu do api. Jeśli ca ma być poza siecią dockerową to tworzenie endpointów ułatwiających generowanie certów jest zbędne.